include/nbl/system/CFileView.h

```diff
@@ -0,0 +1,54 @@
+#ifndef C_MEMORY_FILE_H
+#define C_MEMORY_FILE_H
+#include <nbl/system/IFile.h>
+
+namespace nbl::system
+{
+class CFileView : public IFile
+{
+public:
+	CFileView(const std::filesystem::path& _name, std::underlying_type_t<E_CREATE_FLAGS> _flags) : IFile(nullptr,_flags | ECF_COHERENT | ECF_MAPPABLE), m_name(_name)
+	{
+	}
+
+	virtual const std::filesystem::path& getFileName() const override
+	{
+		return m_name;
+	}
+
+	void* getMappedPointer() override final { return m_buffer.data(); }
+	const void* getMappedPointer() const override final { return m_buffer.data(); }
+
```

Comment:
leave a TODO: comment about needing to figure out how mapped files should grow/shrink